### PR TITLE
Update to Apache Spark 3.0.0 with Python 3.6 and remove Py 2.7 support

### PR DESCRIPTION
**Pull request by tmckayus**
This change moves Apache Spark to version 3.0.0 and at the same
time removes support for Python 2.7. Going forward only Python
3.x will be supported.  Additionally, the base image has been
changed to centos8.

### CODE REVIEW

**Comment by tmckayus** [comment]
note, I haven't tested any of this yet beyond just the build. Expecting the CI tests to kick in here.

File: image.yaml [hunk 1]
```diff
@@ -4,41 +4,33 @@ schema_version: 1
 # to give a meaningful error on an attempt to install Spark in
 # a complete image via S2I
 
-version: 2.4-latest
+version: 2.4.6
```
Comment (tmckayus) [inline image.yaml:7]: need to change this to 3.0.0

**Comment by elmiko** [comment]
hey Trevor, this looks cool!

i will try to give a review, but no promises ;)

File: image.yaml [hunk 2]
```diff
@@ -4,41 +4,33 @@ schema_version: 1
...
-    - name: SPARK_INSTALL
-      value: /opt/spark-distro
-    - name: STI_SCRIPTS_PATH
-      value: /usr/libexec/s2i
-    - name: SCL_ENABLE_CMD
-      value: ""
+
+packages:
+    install:
+        - java-1.8.0-openjdk
```
Comment (tmckayus) [inline image.yaml:21]: should check on this for spark 3

**Comment by tmckayus** [comment]
Note, I tried switching over to podman exclusively but that requires ubuntu 18.04 which then causes errors with docker and oc cluster up, so instead I just made podman vs docker selectable and used docker in travis ...

Someone with more time someday could maybe move our travis tests off of trusty :)

File: image.yaml (hunk 2 above)
Comment (rimolive) [reply image.yaml:21]: Definitely we should move to java 11 for two reasons:

- Java 8 is EOL by Oracle for a while, although Red Hat will support until 2026
- Java 11 has better support in containerized environments

**Comment by tmckayus** [comment]
Okay, this seems to work. I tested it using a python s2i with spark 3 and ran SparkPi and a slightly modified grafzahl (changed for python 3) and it worked.

Summary:
Python 3.6
Spark 3.0.0
java-11-openjdk

Also the repo was updated for cekit 3.6

**Pull request merged**